templates/repo/commit_page.tmpl

```diff
@@ -18,14 +18,123 @@
+			<div class="df mb-4">
+				<h3 class="mb-0 f1"><span class="commit-summary" title="{{.Commit.Summary}}">{{RenderCommitMessage $.Context .Commit.Message $.RepoLink $.Repository.ComposeMetas}}</span>{{template "repo/commit_statuses" dict "Status" .CommitStatus "Statuses" .CommitStatuses  "root" $}}</h3>
+				{{if not $.PageIsWiki}}
+					<div class="ui">
+						<a class="ui blue tiny button" href="{{.SourcePath}}">
+							{{.i18n.Tr "repo.diff.browse_source"}}
+						</a>
+						{{if and ($.Permission.CanWrite $.UnitTypeCode) (not $.Repository.IsArchived) (not .IsDeleted)}}{{- /* */ -}}
+							<div class="ui blue tiny floating dropdown icon button">{{.i18n.Tr "repo.commit.actions"}}
+								{{svg "octicon-triangle-down" 14 "dropdown icon"}}<span class="sr-mobile-only">{{.i18n.Tr "repo.commit.actions"}}</span>
+								<div class="menu">
+									<div class="ui header">{{.i18n.Tr "repo.commit.actions"}}</div>
+									<div class="divider"></div>
+									<div class="item show-create-branch-modal"
+										data-content="{{$.i18n.Tr "repo.branch.new_branch_from" (.CommitID)}}"
+										data-branch-from="{{ShortSha .CommitID}}"
+										data-branch-from-urlcomponent="{{.CommitID}}"
+										data-modal="#create-branch-modal">
+										{{.i18n.Tr "repo.branch.create_branch_operation"}}
+									</div>
+									<div class="item show-create-branch-modal"
+										data-content="{{$.i18n.Tr "repo.branch.new_branch_from" (.CommitID)}}"
+										data-branch-from="{{ShortSha .CommitID}}"
+										data-branch-from-urlcomponent="{{.CommitID}}"
+										data-modal="#create-tag-modal"
+										data-modal-from-span="#modal-create-tag-from-span"
+										data-modal-form="#create-tag-form">
+										{{.i18n.Tr "repo.tag.create_tag_operation"}}
+									</div>
+									<div class="item show-modal revert-button"
+										data-modal="#cherry-pick-modal"
+										data-modal-cherry-pick-type="revert"
+										data-modal-cherry-pick-header="{{$.i18n.Tr "repo.commit.revert-header" (ShortSha .CommitID)}}"
+										data-modal-cherry-pick-content="{{$.i18n.Tr "repo.commit.revert-content"}}"
+										data-modal-cherry-pick-submit="{{.i18n.Tr "repo.commit.revert"}}">{{.i18n.Tr "repo.commit.revert"}}</a></div>
+									<div class="item cherry-pick-button show-modal"
+										data-modal="#cherry-pick-modal"
+										data-modal-cherry-pick-type="cherry-pick"
+										data-modal-cherry-pick-header="{{$.i18n.Tr "repo.commit.cherry-pick-header" (ShortSha .CommitID)}}"
+										data-modal-cherry-pick-content="{{$.i18n.Tr "repo.commit.cherry-pick-content"}}"
+										data-modal-cherry-pick-submit="{{.i18n.Tr "repo.commit.cherry-pick"}}">{{.i18n.Tr "repo.commit.cherry-pick"}}</a></div>
```

Comment:
These `data-xxx` can be replaced by `PageData`, then the JS code can get all data directly.

Reply:
This is for data-modal I'm not sure that this is appropriate for PageData

Are you suggesting that we need to change all 54 data-modal to use PageData? Even when they're purely presentational?